### ISSUE TITLE
Parsed time zone wrong when adjusted at runtime

### ISSUE THREAD

**Issue by twaugh** [dateutil/dateutil#163]
Because `tz.tzlocal` works out its offset at import time, there are problems when the time zone is adjusted at runtime.

We are running a test case for a time parsing function based on `dateutil` that parametrizes the local time zone over several values, e.g. UTC, CET.

When running the UTC test, with the TZ environment variable set to UTC and `tzset()` called, a time string indicating UTC (actually 'Z') is parsed. In this instance, the returned object has tzinfo set to `tzlocal()` because the parsed timezone ('UTC') matches the local timezone.

However, this is only true for the runtime local timezone. It wasn't UTC at import time but CET, which results in the test failing.

Example:

```
$ cat /tmp/testcase.py 
#!/usr/bin/python3
from dateutil import parser
import os
from time import tzset
os.environ['TZ'] = 'UTC'
tzset()
dt = parser.parse('2015-08-24T10:41:00Z')
assert dt.timestamp() == 1440412860.0
$ python3 /tmp/testcase.py
$ TZ=CET python3 /tmp/testcase.py
Traceback (most recent call last):
  File "/tmp/testcase.py", line 8, in <module>
    assert dt.timestamp() == 1440412860.0
AssertionError
```

**Comment by pganssle** [comment]
@twaugh Are you using 2.4.2? Do you have the same issue if you pull the latest version from github? I'm just wondering if this is a duplicate of #100 (which was fixed with PR #109 and #107).

**Comment by twaugh** [comment]
I am. I haven't tried the latest version from github but that's exactly the issue I was running into, with exactly the sort of fix I was hoping for. Thanks!
